### PR TITLE
[Cooldown Maintenance] Added More Spells

### PR DESCRIPTION
**Pull request by DaleHuntGB**
Added:
- Pillar of Frost [Frost Death Knight] - #44 
- Essence Break [Havoc Demon Hunter] - #44 
- Touch of the Magi [Arcane Mage] - #44 
- Bonedust Brew [Windwalker / Brewmaster Monk]
- Shifting Power [Mage] - #44

### CODE REVIEW

**Comment by Tercioo** [comment]
I might be wrong here, but I think the cooldown time threshold to include cooldowns of type 1 (attack cooldowns) is 60 seconds.
Could you check the list and confirm this?

**Comment by DaleHuntGB** [comment]
Okay, after some more testing...

Touch of the Magi shows correctly until CDR is used (for example, Shifting Power), the update in time then works but the ability automatically hides afterwards - as you can see [here](https://i.imgur.com/KfM1Ljm.gif).

Essence Break shows correctly once being used but conditions are not updated once the spell has returned off cooldown. I am using Echo Raid Tools, a simple desaturation is added when the spell is on cooldown and then the desaturation is removed on the ability being available to use but this seems to not be the case on Essence Break. 

Happy to wait for this merge, as more spells are not working. Would reducing the minimum cooldown be possible?

**TL;DR - Minimum CD Threshold for Attack Cooldowns is definitely 60.**

**Comment by Tercioo** [comment]
> Happy to wait for this merge, as more spells are not working. Would reducing the minimum cooldown be possible?

The questions for this issue are:
- When cooldowns below 1 minute are required to track? Not on raid and m+, perhaps arenas?
- Adding down to 45 seconds, would increase the amount of cooldowns added and with that, the data (in bytes) that is required to transmit to other players, would also increase.

**Comment by DaleHuntGB** [comment]
> When cooldowns below 1 minute are required to track? Not on raid and m+, perhaps arenas?

Definitely for arenas, if people find that useful but I would assume most people are looking to mimic the functionality that is provided by OmniCD but might prefer Echo Raid Tools (as I do).

> Adding down to 45 seconds, would increase the amount of cooldowns added and with that, the data (in bytes) that is required to transmit to other players, would also increase.

Completely understand this, and ultimately, it won't be for every use case, I tend to only track the main offensive cooldowns (2mins or more) but Touch of the Magi is a pretty important cooldown as far as burst goes, so that might be useful to track.

**Comment by Tercioo** [comment]
Hi, found a solution.
Could you add the key **arena = true** on cooldowns only available for pvp arena?
I don't know if your current comment has pvp cooldown or it is ready to merge, please let me know.

**Comment by DaleHuntGB** [comment]
I am happy to do that but out of interest, how does this solve the CD durations between less than a minute? Won't tracking still be an issue?

**Comment by Tercioo** [comment]
Hi!
Regular cooldowns with and above 1 minute, will still be sharing as usual.
But, inside an arena, it also shares all cooldowns marked as arena = true.

**Comment by DaleHuntGB** [comment]
Sorry for not updating this. I will do some testing soon (university is manic).

Just out of interest, how does tracking work for Blessing of Freedom, this is a 25s CD but works effectively for tracking.

There are a few spells that I'd like to add but the CD limit being 1min is causing some issues, I appreciate this should be fixed for `arena = true` check, but this would not count for Dungeon/Raid content?

**Comment by Tercioo** [comment]
Communication on this pr has been made with smoke signals :D
Closing has the expansion moved on.
